default-settings.yml

```diff
@@ -140,6 +140,12 @@ mesh:
 #   coordinates (x, y, z); the expression may be constants.
 initial_magnetization: [0, 0, 1]
 
+# Initial time of the simulation. Default is:
+# - the time recorded in the ‘.sol’ file providing the initial
+#   magnetization, if the initial magnetization was defined this way
+# - zero otherwise
+initial_time:
```

Comment:
It might be interesting line 147 to initialize initial_time to null, but how does the yaml parser would handle it ?

Reply:
I just did a small test to see how libyaml-cpp interprets the field, depending on how it is defined or not. Here is the data type as reported by `YAML::Node::Type()`:

```yaml
# (no initial_time)  # YAML::NodeType::Undefined
initial_time:        # YAML::NodeType::Null
initial_time: null   # YAML::NodeType::Null
initial_time: 3e-12  # YAML::NodeType::Scalar
```

Thus, leaving the field empty is the same as setting it to `null`. An empty field is more consistent with the rest of the file, as it is already used for `outputs.file_basename` and `mesh.filename`. [The latter is documented][1] with the note “If this is missing or empty, it defaults to…”.

[1]: https://feellgood.neel.cnrs.fr/reference.html#outputsfile_basename